src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromDelimeter = fromFormat[fromFormat.length - 1];
+  const dateArr = date.split(fromDelimeter);
+  const dataObj = {};
+
+  for (let d = 0; d <= 2; d++) {
+    dataObj[fromFormat[d]] = dateArr[d];
+  }
```

Comment:
This loop assumes that the date parts (year, month, day) are always the first three elements of the `fromFormat` array. If the format ever changes to include other elements or a different order, this will break. It would be more robust to iterate over the `fromFormat` array until the second to last element, which is known to be the delimiter.